File: csharp/demo/demo08a-return-value.cs
/*
 * GETTING RETURNED VALUES FROM THREADS
 */
using System;
using System.Threading;



class Demo08A : IRunnable
{
    public void run()
    {
        int resFoo = 0, resBar = 0;

        var thFoo = new Thread(() => resFoo = doubleValue(5));
        var thBar = new Thread(() => resBar = doubleValue(80));

        thFoo.Start();
        thBar.Start();

        // Wait until thFoo and thBar finish
        thFoo.Join();
        thBar.Join();

        Console.WriteLine(resFoo);
        Console.WriteLine(resBar);
    }


    private int doubleValue(int value)
    {
        return value * 2;
    }
}

File: python/demo08a_return_value.py
'''
GETTING RETURNED VALUES FROM THREADS
'''

import threading



def double_value(value):
    return value * 2



res = {}

th_foo = threading.Thread( target=lambda: res.update({ 'foo': double_value(5) }) )
th_bar = threading.Thread( target=lambda: res.update({ 'bar': double_value(80) }) )

th_foo.start()
th_bar.start()

# Wait until th_foo and th_far finish
th_foo.join()
th_bar.join()

print(res['foo'])
print(res['bar'])
